src/main/java/io/pravega/connectors/flink/Pravega.java

```diff
@@ -0,0 +1,332 @@
+        properties.putLong(CONNECTOR_READER_READER_GROUP_REFRESH_INTERVAL, readerGroupInfo.getReaderGroupConfig().getGroupRefreshTimeMillis());
+        properties.putLong(CONNECTOR_READER_READER_GROUP_EVENT_READ_TIMEOUT_INTERVAL, tableSourceReaderBuilder.eventReadTimeout.toMilliseconds());
+        properties.putLong(CONNECTOR_READER_READER_GROUP_CHECKPOINT_INITIATE_TIMEOUT_INTERVAL, tableSourceReaderBuilder.checkpointInitiateTimeout.toMilliseconds());
+    }
+
+    public TableSourceReaderBuilder tableSourceReaderBuilder() {
+        tableSourceReaderBuilder = new TableSourceReaderBuilder();
+        return tableSourceReaderBuilder;
+    }
+
+    public TableSinkWriterBuilder tableSinkWriterBuilder() {
+        tableSinkWriterBuilder = new TableSinkWriterBuilder();
+        return tableSinkWriterBuilder;
+    }
+
+    /**
+     * Reader builder which can be used to define the Pravega reader configurations. The supplied configurations will be used
+     * to create appropriate Table source implementation.
+     *
+     */
+    public static class TableSourceReaderBuilder<T extends AbstractStreamingReaderBuilder>
+            extends AbstractStreamingReaderBuilder<Row, TableSourceReaderBuilder> {
+
+        private DeserializationSchema<Row> deserializationSchema;
+
+        @Override
+        protected DeserializationSchema<Row> getDeserializationSchema() {
+            return this.deserializationSchema;
+        }
+
+        @Override
+        protected TableSourceReaderBuilder builder() {
+            return this;
+        }
+
+        /**
+         * Pass the deserialization schema to be used.
+         * @param deserializationSchema the deserialization schema.
+         * @return TableSourceReaderBuilder instance.
+         */
+        protected TableSourceReaderBuilder withDeserializationSchema(DeserializationSchema<Row> deserializationSchema) {
+            this.deserializationSchema = deserializationSchema;
+            return this;
+        }
+
+        /**
+         * factory to build an {@link FlinkPravegaInputFormat} for using Table API in a batch environment.
+         * @return a supplier to eagerly validate the configuration and lazily construct the input format.
+         */
+        FlinkPravegaInputFormat<Row> buildInputFormat() {
+            Preconditions.checkState(deserializationSchema != null, "The deserializationSchema must be provided.");
+            final List<StreamWithBoundaries> streams = resolveStreams();
+            final ClientConfig clientConfig = getPravegaConfig().getClientConfig();
+            return new FlinkPravegaInputFormat<>(clientConfig, streams, deserializationSchema);
+        }
+    }
+
+    /**
+     * Writer builder which can be used to define the Pravega writer configurations. The supplied configurations will be used
+     * to create appropriate Table sink implementation.
+     *
+     */
+    public static class TableSinkWriterBuilder<T extends AbstractStreamingWriterBuilder>
```

Comment:
Could the `@Builder` lombok annotation help here to reduce boilerplate code? (maybe in other places as well)

Reply:
Since we are reusing the existing builders (the ones that are used for `FlinkPravegaReader` and `FlinkPravegaWriter`), a lombok builder is not essential in this context